src/common/snippets/src/lowered/pass/domain_optimization.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/lowered/pass/domain_optimization.hpp"
+
+#include "snippets/itt.hpp"
+#include "snippets/lowered/linear_ir.hpp"
+#include "snippets/snippets_isa.hpp"
+#include "snippets/shape_inference/shape_inference.hpp"
+
+
+namespace ov {
+namespace snippets {
+namespace lowered {
+namespace pass {
+
+DomainOptimization::DomainOptimization(size_t& tile_rank) : Pass(), m_tile_rank(tile_rank) {
+}
+bool DomainOptimization::optimize(std::vector<VectorDims>& input_shapes,
+                                  VectorDims& master_shape,
+                                  const size_t total_work_amount,
+                                  const size_t min_parallel_work_amount,
+                                  const size_t min_jit_work_amount) {
+    if (master_shape.size() <= 2)
+        return false;
+
+    auto CollapseLastDim = [](VectorDims& dims) {
+        OPENVINO_ASSERT(dims.size() >= 2, "CollapseLastDim can't process shape with less than two dims");
+        dims[dims.size() - 2] *= dims.back();
+        for (auto i = dims.size() - 1; i > 0; i--)
```

Comment:
Why 🔥 ? 🙃 
`dims.size()` is guaranteed to be  >= 2, so no `size_t` overflow could happen here. Could you elaborate on your concerns here please?

Reply:
BTW casted dims.size() to int to eliminate any concerns

Reply:
I'm simple person: if I see reverse iteration in loop, set `int` to avoid saturation nearby zero :smile:
But also now I see that there is condition `i > 0`, so we never reach the saturation here. Good.